### PR TITLE
editoast: activate 'datadog' feature on production images

### PR DESCRIPTION
**Pull request by woshilapin**
Our default deployment need Datadog so let's make it an option for building production images.

### CODE REVIEW

File: editoast/Dockerfile [hunk 1]
```diff
@@ -46,13 +46,14 @@ COPY --from=planner /editoast/recipe.json recipe.json
 COPY --from=planner /editoast/editoast_derive/ editoast_derive
 ENV RUSTFLAGS="-C link-arg=-fuse-ld=mold"
 ARG CARGO_PROFILE=release
+ARG CARGO_FEATURES
 RUN --mount=type=cache,target=/usr/local/cargo/registry \
     --mount=type=cache,target=/editoast/target \
-    cargo chef cook --profile ${CARGO_PROFILE} --recipe-path recipe.json
+    cargo chef cook --features ${CARGO_FEATURES} --profile ${CARGO_PROFILE} --recipe-path recipe.json
```
Comment (emersion) [inline editoast/Dockerfile]: Hm. `${}` is docker-style expansion, so I'm worried this gets expanded to `cook --features  --profile debug` for local builds. Does this work fine still?

Comment (woshilapin) [reply editoast/Dockerfile]: Oh, right... why didn't I thought about that (thank you peer review 😄). It probably doesn't work indeed. Should be fix in the last push.

**Review by emersion — APPROVED** [review]

**Review by leovalais — APPROVED** [review]

**Review by flomonster — APPROVED** [review]

**Review by ghost — APPROVED** [review]
Lgtm but not tested

**Review by younesschrifi — APPROVED** [review]

**Review by Khoyo — APPROVED** [review]
LGTM. Is the feature unnecessary to compute the recipe.json file?

**Comment by woshilapin** [comment]
> LGTM. Is the feature unnecessary to compute the recipe.json file?

The whole point of the recipe is to build as many dependencies as possible to put in a image layer as a cache so we don’t compile it later. So I thought yes, it makes sense to compile this optional dependency in the recipe. Do you have something different in mind?

**Comment by Khoyo** [comment]
> > LGTM. Is the feature unnecessary to compute the recipe.json file?
> 
> The whole point of the recipe is to build as many dependencies as possible to put in a image layer as a cache so we don’t compile it later. So I thought yes, it makes sense to compile this optional dependency in the recipe. Do you have something different in mind?

I was wondering if cargo chef included all feature-gated dependencies in the recipe.json file or not.

**Pull request merged**